core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -4664,9 +4664,6 @@ private void analyzeRowFilter(String currentIdentity, Table table, QualifiedObje
             ExpressionAnalysis expressionAnalysis;
             try {
                 Identity filterIdentity = filter.getSecurityIdentity()
-                        .map(filterUser -> Identity.forUser(filterUser)
-                                .withGroups(groupProvider.getGroups(filterUser))
-                                .build())
```

Comment:
You have to be careful here, because group are managed by Trino, not access controls. So you still want to create a new `Identity` here with groups from the group provider. (Same for masks and constraints.)

There should be tests for this, too.

Reply:
Doh, good point; added back the `withGroups` phrase.

Since all tests passed even though I had incorrectly removed the `.withGroups` phrase, I guess that means we don't have any tests for that 😞